packages/features/vault/src/side-drawers/deposit-collateral/deposit-collateral.tsx

```diff
@@ -1,48 +1,45 @@
+import { useContext } from 'react';
 import { WalletDepositInput } from '@notional-finance/trade';
 import { VAULT_ACTIONS } from '@notional-finance/shared-config';
-import { useParams } from 'react-router-dom';
 import { VaultSideDrawer } from '../components/vault-side-drawer';
+import { VaultActionContext } from '../../vault-view/vault-action-provider';
 import { useDepositCollateral } from './use-deposit-collateral';
 import { messages } from '../messages';
 
-interface VaultParams {
-  vaultAddress: string;
-  sideDrawerKey?: VAULT_ACTIONS;
-}
-
 export const DepositCollateral = () => {
-  const { vaultAddress } = useParams<VaultParams>();
-
   const {
-    canSubmit,
-    transactionData,
-    depositError,
-    targetLeverageRatio,
-    primaryBorrowSymbol,
-    updatedVaultAccount,
-    updateDepositCollateralState,
-  } = useDepositCollateral(vaultAddress);
+    updateState,
+    state: {
+      primaryBorrowSymbol,
+      updatedVaultAccount,
+      vaultAddress,
+      buildTransactionCall,
+    },
+  } = useContext(VaultActionContext);
+  const transactionData = useDepositCollateral();
+  const canSubmit = buildTransactionCall ? true : false;
+  const currentVaultAddress = vaultAddress || '';
 
   return (
     <VaultSideDrawer
       action={VAULT_ACTIONS.DEPOSIT_COLLATERAL}
       canSubmit={canSubmit}
```

Comment:
vaultAction, canSubmit, vaultAddress and updatedVaultAccount can all come from the context directly, maybe we can just put them inside the VaultSideDrawer component directly?